managed/services/inventory/agents.go

```diff
@@ -680,42 +701,48 @@ func (as *AgentsService) AddProxySQLExporter(ctx context.Context, p *inventoryv1
 }
 
 // ChangeProxySQLExporter updates proxysql_exporter Agent with given parameters.
-func (as *AgentsService) ChangeProxySQLExporter(ctx context.Context, req *inventoryv1.ChangeProxySQLExporterRequest) (*inventoryv1.ProxySQLExporter, error) {
-	agent, err := as.changeAgent(req.AgentId, req.Common)
+func (as *AgentsService) ChangeProxySQLExporter(ctx context.Context, p *inventoryv1.ChangeProxySQLExporterParams) (*inventoryv1.ChangeAgentResponse, error) {
+	ag, err := as.changeAgent(p.AgentId, p.Common)
 	if err != nil {
 		return nil, err
 	}
 
-	res := agent.(*inventoryv1.ProxySQLExporter) //nolint:forcetypeassert
-	as.state.RequestStateUpdate(ctx, res.PmmAgentId)
+	agent := ag.(*inventoryv1.ProxySQLExporter) //nolint:forcetypeassert
+	as.state.RequestStateUpdate(ctx, agent.PmmAgentId)
+
+	res := &inventoryv1.ChangeAgentResponse{
+		Agent: &inventoryv1.ChangeAgentResponse_ProxysqlExporter{
```

Comment:
I believe that we can return only Agent value here, so we don't have to wrap the message in each method

Reply:
These are `proto` types generated by grpc, I don't quite see how we can return arbitrary values here. Let's discuss.

Reply:
we can wrap it in the method where we call this method, as far as I see all these methods with duplicated wrap logic is called by ChangeAgent method

Reply:
I couldn't implement it quite as you requested, but I succeeded in further aligning `AddAgent` with `ChangeAgent` so they now look similar. A lot of redundant code was removed as well.